components/perl/Parse-Yapp/Parse-Yapp-PERLVER.p5m

```diff
@@ -0,0 +1,35 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2020 Rouven Weiler
+#
+
+set name=pkg.fmri value=pkg:/library/perl-5/parse-yapp-$(PLV)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="Perl extension for generating and using LALR parsers"
+set name=info.classification value=org.opensolaris.category.2008:Development/Perl
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license Parse-Yapp.license license="Artistic"
+
+file path=usr/perl5/$(PERLVER)/bin/yapp
```

Comment:
Should there be usr/bin/yapp => usr/perl5/$(PERLVER)/bin/yapp mediated symlink? Is it designed to be used by end user?

Reply:
Concerning the mediator I am new to that. What is the difference between "system-perl" and "perl"?

If "system-perl" means "the actually used one in the system", would that work:
```
link path=usr/bin/yapp target=../perl5/$(PERLVER)/bin/yapp \
    mediator=system-perl mediator-version=$(PERLVER)
```

Reply:
system-perl is one used by illumos-gate scripts, i.e. /usr/perl5/bin/perl. 
perl is usr/bin/perl. The difference is that one can't just change system-perl and expect things to be working. Please, use perl mediator.

Reply:
I looked into libwww-perl.PERLVER.p5m (https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/components/perl/libwww-perl/libwww-perl-PERLVER.p5m). It also adds scripts to /usr/perl5/bin, but does not use a mediator.

Analog to this I removed the mediator for yapp. This then solve this issue here. I think that is okay as it make things transparent.

Reply:
As I understand, yapp is intended to be used by end user, so it makes sense to make it mediated.

Reply:
Just added:
```
link path=usr/bin/yapp target=../perl5/$(PERLVER)/bin/yapp \
    mediator=perl mediator-version=$(PERLVER)
```
I hope I understood correctly to use `mediator=perl` instead of `mediator=system-perl`.

>